.github/workflows/pkgci_test_sharktank.yml

```diff
@@ -64,9 +64,8 @@ jobs:
       - name: Run Sharktank models test suite
         run: |
           source ${VENV_DIR}/bin/activate
-          pytest iree-test-suites/sharktank_models/ \
+          HIP_TARGET=gfx1100 pytest iree-test-suites/sharktank_models/ \
               -rA \
-              -m "target_cpu" \
```

Comment:
There are two things that should vary here based on the `matrix`:

1. the environment variables
2. the pytest filters

How about something like this?

```yml
      matrix:
        include:
          # CPU
          - name: cpu_llvm_task
            runs-on: ubuntu-20.04
            pytest-extra-args: -m "target_cpu"
          # AMD GPU
          - name: amdgpu_hip_rdna3
            runs-on: nodai-amdgpu-w7900-x86-64
            pytest-extra-args: -m "target_hip"
            hip-target: gfx1100
```

Then here:
```yml
      - name: Run Sharktank models test suite
        run: |
          source ${VENV_DIR}/bin/activate
          pytest iree-test-suites/sharktank_models/ \
              -rA \
              --log-cli-level=info \
              --override-ini=xfail_strict=false \
              --timeout=120 \
              --durations=0 \
              ${{ matrix.pytest-extra-args }}
        env:
          HIP_TARGET: ${{ matrix.hip-target }}
```

---

You could also set environment variables like this:

```yml
    steps:
    # ...
    - name: Set HIP_TARGET
      if: contains(matrix.name, "rdna3")
      run: echo "HIP_TARGET=gfx1100" >> $GITHUB_ENV
```

or add a pytest arg:

```yml
      matrix:
        include:
          # CPU
          - name: cpu_llvm_task
            runs-on: ubuntu-20.04
            pytest-extra-args: -m "target_cpu" --hip-target "gfx1100"
```

but what I suggested above is consistent with some other workflow code we have: https://github.com/iree-org/iree/blob/a30a4193bd58e7562e6b8a730e6f1c1962a71366/.github/workflows/pkgci_regression_test.yml#L139-L147 https://github.com/iree-org/iree/blob/a30a4193bd58e7562e6b8a730e6f1c1962a71366/.github/workflows/pkgci_regression_test.yml#L194-L195